parlai/agents/transformer/modules.py

```diff
@@ -1077,10 +1079,15 @@ def prepare_head(tensor):
         if key is None and value is None:
             # self attention
             key = value = query
+            _, _key_len, dim = query.size()
```

Comment:
hmm, is there a need to define these vars both here and in line 1090 below?

Reply:
Yeah it was to make the typechecker happy.